flink-connector/src/test/java/com/datastrato/gravitino/flink/connector/integration/test/hive/FlinkHiveCatalogIT.java

```diff
@@ -0,0 +1,238 @@
+    Assertions.assertInstanceOf(GravitinoHiveCatalog.class, catalog.get());
+
+    // List catalogs.
+    String[] catalogs = tableEnv.listCatalogs();
+    Assertions.assertEquals(2, catalogs.length, "Should create a new catalog");
+    Assertions.assertTrue(
+        Arrays.asList(catalogs).contains(catalogName), "Should create the correct catalog.");
+
+    Assertions.assertEquals(
+        DEFAULT_CATALOG,
+        tableEnv.getCurrentCatalog(),
+        "Current catalog should be default_catalog in flink");
+
+    // Change the current catalog to the new created catalog.
+    tableEnv.useCatalog(catalogName);
+    Assertions.assertEquals(
+        catalogName,
+        tableEnv.getCurrentCatalog(),
+        "Current catalog should be the one that is created just now.");
+
+    // Drop the catalog. Only support drop catalog by SQL.
+    tableEnv.useCatalog(DEFAULT_CATALOG);
+    tableEnv.executeSql("drop catalog " + catalogName);
+    Assertions.assertFalse(metalake.catalogExists(catalogName));
+
+    Optional<Catalog> droppedCatalog = tableEnv.getCatalog(catalogName);
+    Assertions.assertFalse(droppedCatalog.isPresent(), "Catalog should be dropped");
+  }
+
+  @Test
+  public void testCreateGravitinoHiveCatalogUsingSQL() {
+    tableEnv.useCatalog(DEFAULT_CATALOG);
+
+    // Create a new catalog.
+    String catalogName = "gravitino_hive_sql";
+    tableEnv.executeSql(
+        String.format(
+            "create catalog %s with ("
+                + "'type'='gravitino-hive', "
+                + "'hive-conf-dir'='src/test/resources/flink-tests',"
+                + "'hive.metastore.uris'='thrift://127.0.0.1:9084',"
+                + "'unknown.key'='unknown.value'"
+                + ")",
+            catalogName));
+    Assertions.assertTrue(metalake.catalogExists(catalogName));
+
+    // Check the properties of the created catalog.
+    com.datastrato.gravitino.Catalog gravitinoCatalog = metalake.loadCatalog(catalogName);
+    Map<String, String> properties = gravitinoCatalog.properties();
+    Assertions.assertEquals("thrift://127.0.0.1:9084", properties.get(METASTORE_URIS));
+    Map<String, String> flinkProperties =
+        properties.entrySet().stream()
+            .filter(e -> e.getKey().startsWith(PropertiesConverter.FLINK_PROPERTY_PREFIX))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    Assertions.assertEquals(3, flinkProperties.size());
+    Assertions.assertEquals(
+        "src/test/resources/flink-tests",
+        flinkProperties.get(flinkByPass(HiveCatalogFactoryOptions.HIVE_CONF_DIR.key())));
+    Assertions.assertEquals(
+        GravitinoHiveCatalogFactoryOptions.IDENTIFIER,
+        flinkProperties.get(flinkByPass(CommonCatalogOptions.CATALOG_TYPE.key())));
+    Assertions.assertEquals(
+        "unknown.value",
+        flinkProperties.get(flinkByPass("unknown.key")),
+        "The unknown.key will not cause failure and will be saved in Gravitino.");
```

Comment:
The unknown key will be saved in Gravitino but not be ignored.

Reply:
Why does it works? the `unknow.key` not exists in `requiredOptions` or `optionalOptions` and the customer CatalogFactoryHelper only works when loading catalog from Gravitino. Do I miss something?

Reply:
@FANNG1 When creating the catalog, it will pass the `unknow.key` to the Context of GravitinoHiveCatalog and the CatalogDescription of the GravitinoCatalogStore. The GravitinoCatalogStore will not validate the `unknow.key`. And we use `FactoryUtils.GravitinoCatalogFactoryHelper` instead of `FactoryUtil.CatalogFactoryHelper` to skip the validation. So the `unknow.key` will save to the Gravitino successfully. 

Of course, there is a little weird about the key that is not in the `optionalOptions` not failing the Flink Job.

Reply:
@FANNG1 Do you think this is reasonable?

Reply:
it is a little weird,but let's go ahead